Task 4/Task 4-1/Task 4-1.cpp

```diff
@@ -0,0 +1,192 @@
+﻿#include <iostream>
+#include "random"
+
+using namespace std;
+
+/**
+ * \brief Класс переменных метода ввода
+ */
+enum class INPUT 
+{
+USER,
+COMP
+};
+
+/**
+ * \brief Функция выводит заданное сообщение и позволяет пользователю ввести чило
+ * \param str Строка для вывода
+ * \return Введённое число
+ */
+int Input_integer(const string str);
+/**
+ * \brief Функция автоматически заполняет массив целыми числами от -1000 до 1000
+ * \param array Передаваемый массив
+ * \param size Размера массива
+ */
+void Auto_fill(int* array, const size_t size);
+/**
+ * \brief Функция позволяет вручную заполнить массив
+ * \param array Передаваемый массив
+ * \param size Размер массива
+ */
+void Manual_fill(int* array, const size_t size);
+/**
+ * \brief Функция ищет сумму отрицательных элементов, кратных 10-ти
+ * \param array Передаваемый массив
+ * \param size Размер массива
+ * \return Сумму отрицательных элементов кратных 10-ти
+ */
+int Find_neg10_sum(const int array[], const size_t size);
+/**
+ * \brief Функция меняет первые Х элементов на те же, но в обратном порядке
+ * \param array Передаваемый массив
+ * \param size Размер массива
+ */
+void Change_k_elements(int* array, const size_t size);
+/**
+ * \brief Функция проверяет массив на наличие пары соседних элементов, произведение которых равно заданному числу
+ * \param array Передаваемый массив
+ * \param size Размер массива
+ * \return "true" если пара нашлась, "false" если такой пары нет
+ */
+bool Check_pair(int* array, size_t size);
```

Comment:
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.